generic_config_updater/patch_sorter.py

```diff
@@ -1014,6 +1015,165 @@ def create(self, algorithm=Algorithm.DFS):
+        if not(self.patch_wrapper.validate_config_db_patch_has_yang_models(patch)):
+            raise ValueError(f"Given patch is not valid because it has changes to tables without YANG models")
+
+        # Validate target config
+        self.logger.log_info("Validating target config according to YANG models.")
+        if not(self.config_wrapper.validate_config_db_config(target_config)):
+            raise ValueError(f"Given patch is not valid because it will result in an invalid config")
+
+        # Generate list of changes to apply
+        self.logger.log_info("Sorting patch updates.")
+        changes = self.inner_patch_sorter.sort(patch, algorithm)
+
+        return changes
+
+class ConfigSplitter:
+    def __init__(self, config_wrapper, ignore_paths_from_yang_list):
+        self.config_wrapper = config_wrapper
+        self.ignore_paths_from_yang_list = ignore_paths_from_yang_list
+
+    def split_yang_non_yang_distinct_field_path(self, config):
+        empty_tables = self.config_wrapper.get_empty_tables(config)
+        empty_tables_txt = ", ".join(empty_tables)
+        if empty_tables:
+            raise ValueError(f"Given config has empty tables. Table{'s' if len(empty_tables) != 1 else ''}: {empty_tables_txt}")
+
+        # split based on tables with our without YANG models
+        config_with_yang = self.config_wrapper.crop_tables_without_yang(config)
+        config_without_yang = {}
+        for key in config:
+            if key not in config_with_yang:
+                config_without_yang[key] = copy.deepcopy(config[key])
+
+        path_addressing = PathAddressing()
+        # ignore more config from config_with_yang
+        for path in self.ignore_paths_from_yang_list:
+            if not path_addressing.has_path(config_with_yang, path):
+                continue
+            if path == '': # whole config to be ignored
+                return {}, copy.deepcopy(config)
+
+            # Add to config_without_yang from config_with_yang
+            tokens = path_addressing.get_path_tokens(path)
+            add_move = JsonMove(Diff(config_without_yang, config_with_yang), OperationType.ADD, tokens, tokens)
+            config_without_yang = add_move.apply(config_without_yang)
+
+            # Remove from config_with_yang
+            remove_move = JsonMove(Diff(config_with_yang, {}), OperationType.REMOVE, tokens)
+            config_with_yang = remove_move.apply(config_with_yang)
+
+        # Splitting the config based on 'ignore_paths_from_yang_list' can result in empty tables.
+        # Remove empty tables because they are not allowed in ConfigDb
+        config_with_yang_without_empty_tables = self.config_wrapper.remove_empty_tables(config_with_yang)
+        config_without_yang_without_empty_tables = self.config_wrapper.remove_empty_tables(config_without_yang)
+        return config_with_yang_without_empty_tables, config_without_yang_without_empty_tables
+
+class ChangeWrapper:
```

Comment:
>ChangeWrapper [](http://example.com/codeflow?start=6&length=13)

Could you add some code comment to explain the intention of this class? Does design doc cover it?

Reply:
This class is just an implementation detail related to the patch sorter, the design doc does not go that deep

Reply:
Okay with design doc. What does it actually mean by "adjust_changes"? Could you add some code comment?

Reply:
added a comment explaining `adjust_changes`